### PR TITLE
Smarty update

### PR DESCRIPTION
**Pull request by hlohaus**
### 1. Why is this change necessary?

Security and more updates.
Use snippets everywhere.

See: https://github.com/smarty-php/smarty/blob/master/NEW_FEATURES.txt

### 6. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

File: engine/Library/Enlight/Template/Plugins/resource.parent.php [hunk 1]
```diff
@@ -22,82 +23,55 @@
...
+                if (!$exists) {
+                    if ($s->filepath == $_template->parent->source->filepath) {
+                        $exists = true;
+                    }
+                    continue;
                 }
+                $sources[] = $s;
+                $uid .= $s->filepath;
+                break;
+            } catch (SmartyException $e) {
```
Comment (shyim) [inline engine/Library/Enlight/Template/Plugins/resource.parent.php:56]: Would this not catch also template errors silently?

File: tests/Functional/Regressions/Ticket4788Test.php [hunk 2]
```diff
@@ -75,29 +75,6 @@ protected function tearDown()
         Shopware()->Db()->query($sql, $values);
     }
 
-    /**
-     * Test for long description fallback in category listing
-     */
-    public function testArticleLongDescriptionForCategoryListing()
```
Comment (shyim) [inline tests/Functional/Regressions/Ticket4788Test.php]: Is here any reason why this test has been removed?

**Comment by shyim** [comment]
And change the target pls to 5.6 😊

File: engine/Library/Enlight/Template/Plugins/resource.parent.php (hunk 1 above)
Comment (hlohaus) [reply engine/Library/Enlight/Template/Plugins/resource.parent.php]: Now: Nobody will see the message. normally you will get "parent" ressouce not found.
Whats wrong?

**Comment by bneumann97** [comment]
Squash the commits

**Comment by mitelg** [comment]
hey @hlohaus 
thanks for your amazing PR! :blue_heart: 
for a better overview, it would be nice if you could exclude the files, where you only added a blank line for no particular reason :wink:

**Comment by hlohaus** [comment]
:/ i had. It was the cs fixes. Should i revert squashing?

**Comment by mitelg** [comment]
hey @hlohaus 
maybe you use an old version of the cs-fixer. afaik all files in the `engine/Shopware` directory should have the right code style. 
Beside that you should only fix the code style of files, you really worked on :wink:

**Comment by hlohaus** [comment]
Hey @mitelg,

now the travis fail:

check-code-style:
     [exec] Loaded config default from "/home/travis/build/shopware/shopware/.php_cs.dist".
     [exec] .............................................................   61 / 2337 (  3%)
     [exec]    1) shopware/engine/Shopware/Components/Cron/CronJob.php (header_comment)

**Comment by shyim** [comment]
After a rebase it should be gone :)

**Comment by hlohaus** [comment]
Why? the brunch is up to date.
5.6 is broken:
 https://github.com/shopware/shopware/commits/5.6

**Comment by shyim** [comment]
So finally corrected. After a rebase it should be now green :blush:

**Comment by hlohaus** [comment]
Yes, thanks.

**Comment by shyim** [comment]
I tested the branch and found out following issues:

* Inheritance seems to be broken: The Login Page of the Backend shows the revision. This is only shown normally after Login. I guess the block overriding of backend/index/auth.tpl is not working anymore.
* /campaign/index/emotionId/5 - Landingpage throws a Exception
image
* Searching for "20% alle" does not display products
image

**Comment by hlohaus** [comment]
Okay. 1 to 3 fixed.

**Pull request closed without merging**